test/backend/db_tests.js

```diff
@@ -450,6 +451,49 @@ module.exports = function(config, DB) {
         )
 
         test(
+          'locked accounts',
+          function (t) {
+            t.plan(5)
+
+            var lockedAt = Date.now()
+            var unlockCode = hex16()
+            var uid = Buffer(ACCOUNT.uid)
+            var email = Buffer(ACCOUNT.email)
+
+              // set lockedAt
+            return db.lockAccount(uid, { lockedAt: lockedAt, unlockCode: unlockCode })
+              .then(null, function(err) {
+                t.fail('We should not have failed this .lockAccount() request' + String(err))
+              })
+              .then(function(result) {
+                t.deepEqual(result, {}, 'Returned an empty object for lockAccount')
+                return db.account(uid)
+              })
+              .then(function(account) {
+                t.equal(account.lockedAt, lockedAt, 'account should now be locked')
+                return db.unlockCode(uid)
+              })
+              .then(function (_unlockCode) {
+                t.deepEqual(_unlockCode.unlockCode, unlockCode, 'unlockCode should be set')
+
+                // try to unlock the account
+                return db.unlockAccount(uid)
+              })
+              .then(null, function(err) {
+                t.fail('We should not have failed this .unlockAccount() request' + String(err));
```

Comment:
I was not sure of the right way to go here either, I don't mind dropping errors on the ground as long as it's consistent with how the other DB functions operate. The one advantage to throwing an error here is we can show the user an "expired or invalid link" error if the link is no good w/o adding an extra backend call to check the validity of the unlockCode.